hashtypechecker.cpp

```diff
@@ -0,0 +1,122 @@
+#include "hashtypechecker.h"
+
+#include <QtDebug>
+
+HashTypeChecker::HashTypeChecker()
+{
+    connect(&m_john,SIGNAL(finished(int, QProcess::ExitStatus)),this,SLOT(startParsing()));
+    connect (&m_john, SIGNAL(readyReadStandardOutput()), this, SLOT(processOutput())); // connect process signals with your code
+}
+
+HashTypeChecker::~HashTypeChecker()
+{
+    terminate();
+}
+void HashTypeChecker::start(QString& pathToJohn, QString& pathToPwdFile)
+{
+    // We make sure last process is terminated correctly before
+    // loading a new password file.
+    terminate();
+    m_johnOutput.clear();
+    m_johnOutput.append(pathToPwdFile + "\n");
+    m_john.start(pathToJohn + " --show=types " + pathToPwdFile);
+}
+void HashTypeChecker::terminate(bool shouldProcessWorkToDate)
+{
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.terminate();
+    m_john.waitForFinished(500);
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.kill();
+
+    // Process what have been done so far from JohnOutput ..
+    if(shouldProcessWorkToDate)
```

Comment:
finished() call directly startParsing() but, as defined in the .h, terminate(bool shouldProcessWorkToDate = false) has a default parameter. So, right now, startParsing is only called once because of default parameter for all occurences of terminate. The only usage of calling terminate(true) would be if we decide to allows user to cancel it with a button because he thinks it takes too much time processing, he could see the X types we had time to parse before he cancelled. To avoid being called two times, we should just disable the button when finished signal occurs. However, I don't think it's a good feature to put that bouton.